src/transactions/OfferExchange.cpp

```diff
@@ -1077,8 +1077,10 @@ crossOffer(AbstractLedgerTxn& ltx, LedgerTxnEntry& sellingWheatOffer,
         ltxInner.commit();
     }
 
-    offerTrail.push_back(ClaimOfferAtom(accountBID, offerID, wheat,
-                                        numWheatReceived, sheep, numSheepSend));
+    ClaimAtom atom(CLAIM_ATOM_TYPE_V0);
+    atom.v0() = ClaimOfferAtomV0(accountBID.ed25519(), offerID, wheat,
+                                 numWheatReceived, sheep, numSheepSend);
+    offerTrail.push_back(atom);
```

Comment:
nit: I think you need to call `std::move` here as you're using an explicit temp variable

Reply:
Changed to `emplace_back` but `std::move` isn't necessary with the change in https://github.com/stellar/stellar-core/pull/3101#discussion_r661846143.